scrapy/commands/__init__.py

```diff
@@ -104,3 +106,35 @@ def run(self, args, opts):
         Entry point for running commands
         """
         raise NotImplementedError
+
+
+''''
+The BaseRunSpiderCommands class inherits the ScrapyCommand class and it Used for
+performing common functionality between crawl.py and runspider.py
+'''
+
+
+class BaseRunSpiderCommands(ScrapyCommand):
+
+    def add_options(self, parser):
+        ScrapyCommand.add_options(self, parser)
+        parser.add_option("-a", dest="spargs", action="append", default=[],
+                          metavar="NAME=VALUE",
+                          help="set spider argument (may be repeated)")
+        parser.add_option("-o", "--output", metavar="FILE", action="append",
+                          help="dump scraped items into FILE"
+                          + "(use - for stdout)")
```

Comment:
My line length comment also applies here. You should copy these two methods as is from their original location, there should be no need to change their formatting.

Reply:
@jay24rajput Please address this before we can merge. Thanks!

Reply:
@elacuesta I have already covered it [here](https://github.com/scrapy/scrapy/pull/4552/commits/ed4f4f84082f7023734d1200d88e8f1aec57904c) IMO. Please have a look and let me know if anything is missing. Thanks!

Reply:
@elacuesta Done! Is there anything else that needs to be done? Thanks!

Reply:
@elacuesta the Travis CI build is failing. On the local system ```tox scrapy``` is building successfully and along flake8. Can you please help me find where the problem is?

Reply:
Right, I've seen this error in other PRs as well, it seems to be directly related to `pytest-flake8`:
```
.tox/flake8/lib/python3.8/site-packages/pytest_flake8.py:191: in check_file
    app.parse_preliminary_options_and_args(args)
E   AttributeError: 'Application' object has no attribute 'parse_preliminary_options_and_args'
```
This seems safe and unrelated to that, I'll open a new issue to keep track of the build error.

Reply:
Alright so are we good to merge??